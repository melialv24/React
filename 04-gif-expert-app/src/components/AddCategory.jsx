import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({
    onNewCategory
}) => {

    const [ inputValue, setInputValue ] = useState('Amor') 

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()
        if( inputValue.trim().length <= 1 ) return
        onNewCategory( inputValue.trim() )
        setInputValue('')
    }

    return (
        <form aria-label='form' onSubmit={ onSubmit }>
            <input
                type='text'
                placeholder='Buscar gifs'
                value={inputValue}
                onChange= {onInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
