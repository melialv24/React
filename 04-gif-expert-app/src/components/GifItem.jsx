import React from 'react'
import PropTypes from 'prop-types'
import { url } from 'inspector'

export const GifItem = ({ title, url }) => {
    return (
        <div className='card'>
            <img src={url} alt={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifItem.PropTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
