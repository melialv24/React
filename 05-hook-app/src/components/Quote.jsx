import { useState } from "react";
import { useLayoutEffect, useRef } from "react";



export const Quote = ({ quote, author }) => {

    const pRef = useRef()
    const [boxSize, setBoxSize] = useState({
        width: 0,
        height: 0
    })

    useLayoutEffect(() => {

        //el .current es para tomar el valor actual
        const { width, height} = pRef.current.getBoundingClientRect()

        setBoxSize({
            width, 
            height
        })


    }, [quote])

    return (
        <>
            <blockquote 
                className="blockquote text-end"
                style={{ display: 'flex' }}
            >
                <p 
                    className="mb-2"
                    ref={pRef}
                >
                    { quote }
                </p>
                <footer className="blockquote-footer">{ author }</footer>
            </blockquote>

            <code> { JSON.stringify(boxSize) } </code>
        </>
    )
}
