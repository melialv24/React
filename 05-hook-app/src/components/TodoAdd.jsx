import { useForm } from "../hooks"

export const TodoAdd = ({ onNewTodo }) => {

    const {  formState, onInputChange, id, description, done, onResetForm } = useForm({
        id: new Date().getTime(),
        description: '',
        done: 'false'
    })

    const onFormSubmit = (e) => {
        e.preventDefault()
        if (description.length <= 1 ) return 

        const newTodo = {
            id: new Date().getTime(),
            description,
            done: 'false'
        }

        onNewTodo( newTodo )
        onResetForm()
    }

    return (
        <form onSubmit={ onFormSubmit }>
            <input
                type='text'
                placeholder="Â¿que hay que hacer?"
                className="form-control"
                value={description}
                name='description'
                onChange={onInputChange}
            />

            <button 
                type="submit"
                className="btn btn-primary mt-2"
            > 
                Agregar 
            </button>
        </form>
    )
}
