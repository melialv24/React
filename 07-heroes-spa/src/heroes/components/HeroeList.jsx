import { useMemo } from "react";
import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";
import { HeroCard } from "./HeroCard";

export const HeroeList = ({ publisher }) => {
  const dataHeroes = useMemo( () => getHeroesByPublisher(publisher), [publisher]);

  return (
    <div className="row rows-cols1 row-cols-md-3 g-3">
      {dataHeroes &&
        dataHeroes.map((hero) => <HeroCard key={hero.id} {...hero} />)}
    </div>
  );
};
