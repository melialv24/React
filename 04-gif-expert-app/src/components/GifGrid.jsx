import React from 'react'
import { PropTypes } from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category )
    console.log({ images, isLoading });
    /*const [infoImage, setinfoImage] = useState([])

    useEffect( () => {

        getGifs( category )
            .then( newImages => setinfoImage(newImages))
        
    }, [])*/

    return (
        <>
            <h3>{ category }</h3>

            { isLoading ? 
                (
                    <div>
                        <h2>Cargando...</h2>
                    </div>
                    
                )
                :
                (
                    <div className='card-grid'>
                        { images.map( img => (
                            <GifItem key={img.id}
                                {...img}
                            />
                        )) }
                    </div>
                )
            }
            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}