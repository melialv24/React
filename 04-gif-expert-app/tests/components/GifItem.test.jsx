import {render, screen} from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"

describe('Probando el componente GifItem', () => { 7

    const title = 'MELISSA ERES HERMOSA'
    const url = 'http://localhost/PRUEBA%20DE%20URL'

    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<GifItem title={title} url={url} />)
        expect( container ).toMatchSnapshot()
    })

    test('Probar las props que recibe GifItem', () => { 

        render(<GifItem title={title} url={url}/>)
        // Esto mismo lo podemos hacer desestructurando las propiedades del screen
        //expect( screen.getByRole('img').src).toBe( url )

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
    })

})