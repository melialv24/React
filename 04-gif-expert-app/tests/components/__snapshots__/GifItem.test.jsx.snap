// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="MELISSA ERES HERMOSA"
      src="http://localhost/PRUEBA%20DE%20URL"
    />
    <p>
      MELISSA ERES HERMOSA
    </p>
  </div>
</div>
`;
